multiqc/utils/mqc_colour.py

```diff
@@ -51,6 +339,14 @@ def get_colour(self, val, colformat="hex", lighten=0.3):
         rgb_converter = lambda x: max(0, min(1, 1 + ((x - 1) * lighten)))
 
         try:
+            if self.name in mqc_colour_scale.qualitative_scales and isinstance(val, float):
```

Comment:
I think this is all that's needed to hide this behind linting:
```suggestion
            if self.name in mqc_colour_scale.qualitative_scales and isinstance(val, float) and config.lint:
```